scripts/setup_rdma_env.sh

```diff
@@ -0,0 +1,162 @@
+    exit 1
+fi
+
+mkdir -p "$WORKING_DIR"
+
+install_irdma() {
+    echo "Installing irdma driver..."
+    pushd "$WORKING_DIR"
+    wget "https://downloadmirror.intel.com/${IRDMA_DMID}/irdma-${IRDMA_VER}.tgz" -O ./irdma.tar.gz
+    tar -xf ./irdma.tar.gz
+    pushd "./irdma-${IRDMA_VER}"
+    sudo ./build.sh
+    popd
+    popd
+    modprobe irdma
+}
+
+configure_irdma() {
+    echo "Configuring irdma driver..."
+
+    # enable RoCE
+    roce_ena_val=$(grep "options irdma roce_ena=" /etc/modprobe.d/irdma.conf | cut -d "=" -f 2)
+    if [[ -z "$roce_ena_val" ]]; then
+        echo "options irdma roce_ena=1" | sudo tee -a /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    elif [[ "$roce_ena_val" != "1" ]]; then
+        sudo sed -i '/options irdma roce_ena=/s/roce_ena=[0-9]*/roce_ena=1/' /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    fi
+
+    # increase irdma Queue Pair limit
+    limits_sel_val=$(grep "options irdma limits_sel=" /etc/modprobe.d/irdma.conf | cut -d "=" -f 2)
+    if [[ -z "$limits_sel_val" ]]; then
+        echo "options irdma limits_sel=5" | sudo tee -a /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    elif [[ "$limits_sel_val" != "5" ]]; then
+        sudo sed -i '/options irdma limits_sel=/s/limits_sel=[0-9]*/limits_sel=5/' /etc/modprobe.d/irdma.conf
+        sudo dracut -f
+    fi
+}
+
+install_perftest() {
+    echo "Installing perftest dependencies..."
+    sudo apt update
+    sudo apt install -y libibverbs-dev librdmacm-dev libibumad-dev libpci-dev
+    echo "Installing perftest..."
+    pushd "$WORKING_DIR"
+    wget https://github.com/linux-rdma/perftest/releases/download/24.07.0-0.44/perftest-24.07.0-0.44.g57725f2.tar.gz -O perftest.tar.gz
+    tar -xf perftest.tar.gz
+    pushd ./perftest-24.07.0
+    ./autogen.sh
+    ./configure
+    make -j$(nproc)
+    sudo make install
+    popd
+    popd
+}
+
+install_libfabric() {
+    echo "Installing libfabric from the source..."
+    pushd "$WORKING_DIR"
+    git clone --depth 1 --branch v1.22.0 https://github.com/ofiwg/libfabric libfabric
+    pushd libfabric
+    ./autogen.sh
+    ./configure
+    make -j$(nproc)
+    sudo make install
+    sudo ldconfig
+    popd
+    popd
+}
+
+check_roce() {
+    echo -n "Checking RoCE... "
+    roce_ena_val=$(cat /sys/module/irdma/parameters/roce_ena)
+    if [[ "$roce_ena_val" != "1" ]] && [[ "$roce_ena_val" != "65535" ]]; then
+        echo "FAIL: RoCE disabled"
+        exit 1
+    fi
+    echo "OK"
+}
+
+check_mtu() {
+    echo "Checking MTU..."
+    rdma_devices=($(rdma link | grep "LINK_UP" | cut -d " " -f 8))
+    if [[ "${#rdma_devices[@]}" == "0" ]]; then
+        echo "FAIL: No RDMA devices detected"
+        exit 1
+    fi
+    fail=0
+    for device in "${rdma_devices[@]}"; do
+        mtu=$(ip addr | grep "$device:" | cut -d " " -f 5)
+        if [[ "$mtu" == "9000" ]]; then
+            echo "* $device OK"
+        else
+            echo "* $device FAIL: MTU = $mtu (9000 is recommended)"
```

Comment:
If it is recommended then it should not be a fail but a warning.

Reply:
agree, We should fix it on the main branch, but I think we don't have to fix it now.